src/components/Scn/components/ScLink/ScLink.tsx

```diff
@@ -0,0 +1,109 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useClient } from '@components/ClientProvider';
+import { TLinkFormat } from '@components/Scn';
+import { useInView } from '@hooks/useInView';
+import { ScAddr } from 'ts-sc-client';
+
+import { voidElements } from './constants';
+import { StyledScnLink, StyledScTag, StyledScTagLink } from './styled';
+import { useDifferedLinkContent } from './useDifferedLinkContent';
+
+interface IProps {
+  addr: number;
+  contentType?: TLinkFormat | null;
+  content?: string | null;
+}
+
+interface IHtmlNodeToReactProps {
+  node: ChildNode;
+}
+
+// const nodeAttributesToObject = (node: HTMLElement) => {
+//   if (!node.hasAttributes()) return {};
+//   return Array.from(node.attributes).reduce((acc, curr) => {
+//     if(curr.name === "style") {
+//       console.log(node)
+//       return acc;
+//     }
+//     return {
+//       ...acc,
+//       [curr.name]: curr.value,
+//     };
+//   }, {});
+// };
+
+const HtmlNodeToReact = ({ node }: IHtmlNodeToReactProps) => {
+  if (node instanceof Text) return <>{node.textContent}</>;
+  if (node.nodeName === 'SC_ELEMENT') {
```

Comment:
Do you form the `nodeName` yourself or do you define it beforehand, for example, in types?